src/wp-admin/includes/image.php

```diff
@@ -278,70 +282,98 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
+						$saved = $editor->save( $editor->generate_filename( str_replace( 'image/', '', $output_mime_type ) . '-scaled' ), $output_mime_type );
+						if ( ! is_wp_error( $saved && ! empty( $saved ) ) ) {
+							$image_meta['sources'][ $output_mime_type ] = _wp_get_sources_from_meta( $saved );
+						}
 					}
 				} else {
 					// TODO: Log errors.
 				}
 			} else {
-				// TODO: Log errors.
-			}
-		} elseif ( ! empty( $exif_meta['orientation'] ) && 1 !== (int) $exif_meta['orientation'] ) {
-			// Rotate the whole original image if there is EXIF data and "orientation" is not 1.
-
-			$editor = wp_get_image_editor( $file );
-
-			if ( is_wp_error( $editor ) ) {
-				// This image cannot be edited.
-				return $image_meta;
-			}
-
-			// Rotate the image.
-			$rotated = $editor->maybe_exif_rotate();
-
-			if ( true === $rotated ) {
-				// Append `-rotated` to the image file name.
-				$saved = $editor->save( $editor->generate_filename( 'rotated' ) );
-
-				if ( ! is_wp_error( $saved ) ) {
-					$image_meta = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
+				if ( ! empty( $exif_meta['orientation'] ) && 1 !== (int) $exif_meta['orientation'] ) {
+					// Rotate the whole original image if there is EXIF data and "orientation" is not 1.
+
+					// Rotate the image.
+					$rotated = $editor->maybe_exif_rotate();
+
+					// For the default mime type rotated image is named "my_image-rotated.jpg".
+					if ( true === $rotated && 0 === $mime_index ) {
+						// Append `-rotated` to the image file name.
+						$saved = $editor->save( $editor->generate_filename( 'rotated' ) );
+					} else {
+						// For alternate mimes, append -{mime-extension}-rotated to the image file name.
+						$saved = $editor->save( $editor->generate_filename( str_replace( 'image/', '', $output_mime_type ) . '-rotated' ), $output_mime_type );
+					}
 
-					// Update the stored EXIF data.
-					if ( ! empty( $image_meta['image_meta']['orientation'] ) ) {
-						$image_meta['image_meta']['orientation'] = 1;
+					if ( ! is_wp_error( $saved && ! empty( $saved ) ) ) {
+						$image_meta                                 = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
+						$image_meta['sources'][ $output_mime_type ] = _wp_get_sources_from_meta( $saved );
+						// Update the stored EXIF data.
+						if ( ! empty( $image_meta['image_meta']['orientation'] ) ) {
+							$image_meta['image_meta']['orientation'] = 1;
+						}
+					} else {
+						// TODO: Log errors.
 					}
 				} else {
-					// TODO: Log errors.
+					// For the default mime type, only add the file data to the 'sources' array.
+					if ( 0 === $mime_index ) {
+						$image_meta['sources'][ $output_mime_type ] = array(
+							'file'     => wp_basename( $image_meta['file'] ),
+							'filesize' => isset( $image_meta['filesize'] ) ? $image_meta['filesize'] : wp_filesize( $file ),
+						);
```

Comment:
Shouldn't we also use the new `_wp_get_sources_from_meta` here instead? We can probably adjust that function to be slightly more flexible, specifically use `$meta['filesize']` if it is set. Then this could be something like:
```suggestion
						$image_meta['sources'][ $output_mime_type ] = _wp_get_sources_from_meta(
							array_merge(
								$image_meta,
								array( 'path' => $file )
							)
						);
```

Reply:
`meta['filesize']` is only set on the full sized image (and only since recently), thats why I only use it here.

Reply:
made _wp_get_sources_from_meta more flexible and will use meta filesize if available